LayoutTests/http/tests/push-api/subscribe-deny-permissions.html

```diff
@@ -11,10 +11,10 @@
 navigator.serviceWorker.register("resources/subscribe-worker.js", { }).then(async (registration) => {
     try {
         await waitForState(registration.installing, "activated");
-        await testServiceWorkerPermissionState(registration, 'denied');
-        await testDocumentPermissionState(registration, 'denied');
+        await testServiceWorkerPermissionState(registration, 'prompt');
+        await testDocumentPermissionState(registration, 'prompt');
         await testServiceWorkerSubscribe(registration, 'NotAllowedError');
-        await testDocumentSubscribeWithoutUserGesture(registration, 'NotAllowedError', 'User denied push permission');
+        await testDocumentSubscribeWithoutUserGesture(registration, 'NotAllowedError', 'Push notification prompting can only be done from a user gesture.');
```

Comment:
This particular line in the test should probably not be changed after fixing the issue noted above.

Reply:
The test does not have a user gesture, therefore it should not be able to distinguish denied from prompt. Otherwise it defeats the purpose of this file. That said, we should probably add another check with a user gesture and make sure that it behaves correctly

Reply:
Since the test doesn't have a user gesture, having this subscribe API throw a NotAllowedError with a message saying that the permission is denied is a clear violation of the privacy we're trying to enforce in this patch.

It looks like we're testing the "with user gesture" case below already. Therefore, I don't think we even need to add a test case.

Overall, I think my recommendation is to validate using Safari (manually) that the double prompting that Ben is concerned about does not happen.

Reply:
Sounds good!--we checked and it seems to be fine. No changes were made to this.